src/jwplayer/types.d.ts

```diff
@@ -448,7 +448,7 @@ export interface JwPlayerWrapperProps extends JwPlayerContainerId {
 	getDismissed?: () => boolean;
 	config?: PlayerConfig;
 	playerUrl?: string;
-	onReady?: (playerInstance: Player) => void;
+	onReady?: (playerInstance?: Player) => void; // playerInstance for onReady is not needed in the SR scenario it is needed when AdEngine controls the ads
```

Comment:
how about prevent calling `articlePlayerOnReady` when playerInstance is undefined? 🤔

Reply:
do you mean preventing calling it but also moving the showing placeholder logic out of it? it may be a better long-term solution 👍